services/brig/src/Brig/API/Public.hs

```diff
@@ -738,6 +740,40 @@ listUsersByIdsOrHandles self q = do
     byIds :: Local UserId -> [Qualified UserId] -> (Handler r) [Public.UserProfile]
     byIds lself uids = API.lookupProfiles lself uids !>> fedError
 
+-- Similar to listUsersByIdsOrHandles, except that it allows partial successes
+-- using a new return type
```

Comment:
My understanding is that the new field is optional. If that is so, then it's addition makes the response backwards-compatible. I understand that the new endpoint will not throw for unavailable remote backends, but cannot you bake this into the existing handler? The old endpoint versions wouldn't fail anymore, and they might end up providing an additional field in response, but clients should ignore that field by contract because the field wasn't there when Swaggers for the client API versions were finalised.

Reply:
It isn't backwards compatible because it introduces a new layer of structure to the response. Previously it was a top level list with nowhere to add a second list, now it is an object containing at least one list, and optionally a second.

Reply:
Oh, right, it's not backwards compatible in terms of JSON. Nevertheless, I believe the handler can be generalised and reused for both endpoint versions. Or at least factoring out the common bits would be good to have.

Reply:
I've updated these functions to pull out the common bits